oaknational/Oak-Web-Application
# Feat/pupil 999/exclude maths core

### Job story
- retire KS4 Maths core lessons ensuring that lessons in foundation and higher legacy are still available

### Implementation
This is done by implementing actions inside the following pupil queries:
"pupilProgrammeListingQuery", "pupilSubjectListingQuery", "pupilUnitListingQuery", "pupilLessonListingQuery"

Each uses the helper function in src/node-lib/curriculum-api-2023/helpers/overridesAndExceptions.ts

## How to test

On this preview:

1. Go to https://deploy-preview-2881--oak-web-application.netlify.thenational.academy/pupils/programmes/maths-secondary-year-10/options
2. Observe that Core is no longer offered
3. navigate to  https://deploy-preview-2881--oak-web-application.netlify.thenational.academy/pupils/programmes/maths-secondary-year-10-core-l/units
4. observe the expired banner
5. navigate to  https://deploy-preview-2881--oak-web-application.netlify.thenational.academy/pupils/programmes/maths-secondary-year-10-core-l/units/directed-numbers-fe66/lessons
6. observe the expired banner
7. navigate to https://deploy-preview-2881--oak-web-application.netlify.thenational.academy/pupils/programmes/maths-secondary-year-10-core-l/units/directed-numbers-fe66/lessons/adding-directed-numbers-chjk4t/overview
8. observe the expired banner
9. Try other browse journeys and observe that they are uneffected

## Checklist

- [x]  Maths core excluded but other programmes available
- [x] Expired banner on maths core unit, lesson listings and lessons
- [x] other navigation uneffected
